### ISSUE TITLE
Enhance 'no-tabs' rule: New option to only disallow non-indenting tabs

### ISSUE THREAD

**Issue by fabio-silva** [eslint/eslint#10256]
**What rule do you want to change?** 
[no-tabs](https://eslint.org/docs/rules/no-tabs)

**Does this change cause the rule to produce more or fewer warnings?** 
Fewer, as indenting tabs will be allowed.

**How will the change be implemented? (New option, new default behavior, etc.)?** 
New option.

**Please provide some example code that this change will affect:**

```js
    var name =     'Foo'; //We should get a warning for the tab in between, not for the initial one
```

**What does the rule currently do for this code?** 
A warning because there are tabs for indentation purposes (in the beginning) and for non-necessarily-indentation purposes (after the assignment)

**What will the rule do after it's changed?**
Allows indentation tabs using a proper option, but not non-indenting tabs. This could be something like ```allow-indenting```.

**Comment by platinumazure** [comment]
I'll champion this.

**Comment by platinumazure** [comment]
@fabio-silva Thanks for creating this!

Hope it's okay that I tweaked the title a bit-- just wanted to avoid any team member seeing the title in the issue list and thinking this was about making a breaking change to the rule.

**Comment by not-an-aardvark** [comment]
@eslint/eslint-team Anyone else in favor of this proposal? This issue only needs one more :+1: to accept it, otherwise it's probably time to close it.

**Comment by platinumazure** [comment]
Working on this.
